packages/components/managed-data-table/src/managed-data-table.js

```diff
@@ -0,0 +1,142 @@
+  const columns = props.columns.map((column) => ({
+    ...column,
+    isTruncated: props.displaySettings.useDisplaySettings
+      ? props.displaySettings.isWrappingText
+      : column.isTruncated,
+  }));
+
+  return (
+    <Spacings.Stack>
+      <DataTableSettings
+        topBar={props.topBar}
+        onTableSettingsChange={props.onTableSettingsChange}
+        columnManager={props.columnManager}
+        displaySettings={props.displaySettings}
+      />
+      <DataTable
+        rows={props.rows}
+        columns={columns}
+        footer={props.footer}
+        onRowClick={props.onRowClick}
+        maxHeight={props.maxHeight}
+        maxWidth={props.maxWidth}
+        isCondensed={props.displaySettings.isCondensed}
+        disableHeaderStickiness={props.disableHeaderStickiness}
+        disableFooterStickiness={props.disableFooterStickiness}
+        itemRenderer={props.itemRenderer}
+        isHeaderSticky={props.isHeaderSticky}
+        wrapHeaderLabels={props.wrapHeaderLabels}
+        verticalCellAlignment={props.verticalCellAlignment}
+        horizontalCellAlignment={props.horizontalCellAlignment}
+        sortedBy={props.sortedBy}
+        onSortChange={props.onSortChange}
+        sortDirection={props.sortDirection}
+      />
+    </Spacings.Stack>
+  );
+};
+
+ManagedDataTable.propTypes = {
+  rows: PropTypes.arrayOf(
+    PropTypes.shape({
+      id: PropTypes.string.isRequired,
+    })
+  ).isRequired,
+  columns: PropTypes.arrayOf(
+    PropTypes.shape({
+      key: PropTypes.string.isRequired,
+      /* defaults to `auto` */
+      width: PropTypes.string,
+      label: PropTypes.node.isRequired,
+      align: PropTypes.oneOf(['left', 'center', 'right']),
+      onClick: PropTypes.func,
+      /* custom item renderer, specific for items of this column */
+      renderItem: PropTypes.func,
+      isTruncated: PropTypes.bool,
+      isSortable: PropTypes.bool,
+      disableResizing: PropTypes.bool,
+      shouldIgnoreRowClick: PropTypes.bool,
+    })
+  ).isRequired,
+  footer: PropTypes.node,
+  maxWidth: PropTypes.number,
+  maxHeight: PropTypes.number,
+  onRowClick: PropTypes.func,
+  disableHeaderStickiness: PropTypes.bool,
+  disableFooterStickiness: PropTypes.bool,
+  /* the default item (cell) renderer.
+  an existing per-column `renderItem` func takes precedence over this */
+  itemRenderer: PropTypes.func.isRequired,
+  /* the default cell alignment
+  an existing per-column `align` property takes precedence over this */
+  isHeaderSticky: PropTypes.bool,
+  wrapHeaderLabels: PropTypes.bool,
+  verticalCellAlignment: PropTypes.oneOf(['top', 'center', 'bottom']),
+  horizontalCellAlignment: PropTypes.oneOf(['left', 'center', 'right']),
+  /* Sorting props: */
+  sortedBy: PropTypes.string,
+  onSortChange: PropTypes.func,
+  sortDirection: PropTypes.oneOf(['desc', 'asc']),
+  /* Table settings props */
+  displaySettings: PropTypes.shape({
+    useDisplaySettings: PropTypes.bool,
```

Comment:
Should be name these `enableDisplaySettings` or `disableDisplaySettings`? Personally I like disable more depending on the use case.